firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -126,101 +127,133 @@ String getName() {
 
   /**
-   * Create firebase installation id for the {@link FirebaseApp} on FIS Servers and client side
-   * cache.
+   * Get firebase installation id for the {@link FirebaseApp}. Either from the persisted fid entry
+   * if already exists and is not error state or create a new Fid on the FIS Servers and persist in
+   * the shared prefs.
    *
    * <pre>
    *     The workflow is:
-   *         check if cache empty or cache status is REGISTER_ERROR
+   *          Persisted Fid does not exists Or Persisted Fid is in Error state
    *                                 |
-   *                           create random fid
-   *                                 |
-   *               update cache with cache status UNREGISTERED
-   *                                 |
-   *                           send http request to backend
-   *                                 |                  |
-   *  on success: set cache entry status to REGISTERED  |
-   *                                 |                  |
-   *                                 |         on failure: set cache entry status to REGISTER_ERROR
-   *                                 |                     |
-   *                                return          throw exception
+   *             return a newly created random Fid and persist the Fid entry
+   *
+   *
+   *
+   *                    Persisted Fid exists And is Registered
+   *                                  |
+   *                          return the Persisted Fid
+   *
+   *
    *
+   *                  Persisted Fid exists and is in Unregistered state
+   *                                     |
+   *                Register the Fid on FIS backened and update the PersistedFidEntry
+   *                                     |
+   *                            return the Persisted Fid
    *
-   *                      else if cached FID exists
-   *                             |            |
-   *     if cache status is UNREGISTERED      |
-   *                  |                   return cached FID
-   *     send http request to backend
-   *               |               |
-   *  on success: set cache        \
-   *  entry status to REGISTERED   \
-   *                |              \
-   *                |         on failure: set cache entry status to REGISTER_ERROR
-   *                |                     |
-   *              return          throw exception
    * </pre>
    */
   @WorkerThread
-  private String createFirebaseInstallationId() throws FirebaseInstallationsException {
+  private String getFirebaseInstallationId() throws FirebaseInstallationsException {
 
-    FiidCacheEntryValue cacheEntryValue = localCache.readCacheEntryValue();
+    PersistedFidEntryValue persistedFidEntryValue = persistedFid.readPersistedFidEntryValue();
+
+    if (persistedFidDoesNotExistsOrInErrorState(persistedFidEntryValue)) {
+      return createFidAndPersistFidEntry();
+    }
+
+    if (persistedFidExistsAndRegistered(persistedFidEntryValue)) {
```

Comment:
Can we please move this case to the top, because this is the standard case.

Reply:
Also, something we added to the Web SDK:
We should check here if the current FIS Auth Token is still valid.
If not, this request should trigger an asynchronous request to FIS to update the local FIS Auth Token.

Reply:
As discussed offline, will add this when auth token method is implemented.